lib/ansible/modules/cloud/azure/GUIDELINES.md

```diff
@@ -0,0 +1,155 @@
+
+# Rules for Azure Ansible facts modules
```

Comment:
How about  call it coding guideline. and  add some background context here,  such as it's based on discussion between  MS and  Ansible owner, and will be used as coding  reviewing rules.